rdflib/plugins/sparql/parser.py

```diff
@@ -81,7 +81,7 @@ def expandTriples(terms: ParseResults) -> List[Any]:
                 if len(t) > 1:
                     res += t  # Don't update last_subject/last_predicate
                 # is this bnode the subject of more triples?
-                if i + 1 < l_ and terms[i + 1] not in ".,;":
+                if i + 1 < l_ and not any(terms[i + 1] == token for token in ".,;"):
```

Comment:
I've tried to interpret this code change several times, and it seems to be the logic of the pre-fix state and post-fix state is functionally identical.
1) If this term is not any of the characters in this string, _vs_
2) None of the characters in this string are my term.

Clearly it has fixed the issue, the tests are passing, but I'm not clear on _how_ the change works.

Reply:
The issue is that the terms are not always strings. The __eq__ method on CompValue successfully returns False in this case, but the __contains__ method on str apparently raises a type error, although I can’t find any documentation of this behaviour. Compare with the similar test on line 72.